src/components/Creatable/Creatable.tsx

```diff
@@ -131,6 +131,16 @@ function CreatableComponent({
   const tagsToShow = finalExternalInitialValues ? [...initialTagItems, ...tagItems] : [...tagItems];
   const disableInputs = disabled || (max && tagsToShow.length >= max && max > 0);
 
+  // Check if key value pair has colon
+  const keyInputHasColon = keyValue?.includes(":");
+  const valueInputHasColon = value?.includes(":");
+  const isKeyInputValid = invalid || keyInputHasColon;
+  const isValueInputValid = invalid || valueInputHasColon;
+  const keyInputInvalidText = keyInputHasColon ? "Colon is not allowed" : invalidText;
+  const valueInputInvalidText = valueInputHasColon ? "Colon is not allowed" : invalidText;
+
+  const isAddButtonDisabled = disabled || !existValue || invalid || isKeyInputValid || isValueInputValid;
```

Comment:
You can remove the || invalid on `isAddButtonDisabled` since this check is already present on `isKeyInputValid` and `isValueInputValid`.